src/Microsoft.Data/DataFrame.cs

```diff
@@ -33,6 +36,273 @@ public DataFrame(IList<BaseColumn> columns)
+                        }
+                        break;
+                    case ArrowTypeId.Int64:
+                        ReadOnlyMemory<byte> longValueBuffer = (arrowArray as PrimitiveArray<long>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> longNullBitMapBuffer = (arrowArray as PrimitiveArray<long>).NullBitmapBuffer.Memory;
+                        dataFrameMemory = new Memory<byte>(new byte[longValueBuffer.Length]);
+                        nullBitMapMemory = new Memory<byte>(new byte[longNullBitMapBuffer.Length]);
+                        copied = longValueBuffer.TryCopyTo(dataFrameMemory) && longNullBitMapBuffer.TryCopyTo(nullBitMapMemory);
+                        if (copied)
+                        {
+                            dataFrameColumn = new PrimitiveColumn<long>(field.Name, dataFrameMemory, nullBitMapMemory, arrowArray.Length);
+                        }
+                        break;
+                    case ArrowTypeId.UInt8:
+                    case ArrowTypeId.UInt16:
+                    case ArrowTypeId.UInt32:
+                        ReadOnlyMemory<byte> uintValueBuffer = (arrowArray as PrimitiveArray<uint>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> uintNullBitMapBuffer = (arrowArray as PrimitiveArray<uint>).NullBitmapBuffer.Memory;
+                        dataFrameMemory = new Memory<byte>(new byte[uintValueBuffer.Length]);
+                        nullBitMapMemory = new Memory<byte>(new byte[uintNullBitMapBuffer.Length]);
+                        copied = uintValueBuffer.TryCopyTo(dataFrameMemory) && uintNullBitMapBuffer.TryCopyTo(nullBitMapMemory);
+                        if (copied)
+                        {
+                            dataFrameColumn = new PrimitiveColumn<uint>(field.Name, dataFrameMemory, nullBitMapMemory, arrowArray.Length);
+                        }
+                        break;
+                    case ArrowTypeId.UInt64:
+                        ReadOnlyMemory<byte> ulongValueBuffer = (arrowArray as PrimitiveArray<ulong>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> ulongNullBitMapBuffer = (arrowArray as PrimitiveArray<ulong>).NullBitmapBuffer.Memory;
+                        dataFrameMemory = new Memory<byte>(new byte[ulongValueBuffer.Length]);
+                        nullBitMapMemory = new Memory<byte>(new byte[ulongNullBitMapBuffer.Length]);
+                        copied = ulongValueBuffer.TryCopyTo(dataFrameMemory) && ulongNullBitMapBuffer.TryCopyTo(nullBitMapMemory);
+                        if (copied)
+                        {
+                            dataFrameColumn = new PrimitiveColumn<ulong>(field.Name, dataFrameMemory, nullBitMapMemory, arrowArray.Length);
+                        }
+                        break;
+                    case ArrowTypeId.Binary:
```

Comment:
I *think* this case is wrong. Binary has 3 buffers in Arrow, so I need to look into this further